### PR TITLE
Iatr/circular

### LINKED ISSUE #48: Circularized File exception check improve
Currently the Circularized file errors are checked and the default duration is 2 minutes. However this may cause repeated errors when the last read was successful -- this is because the check is always for two minutes . The check should be not for the last 2 minuets but since the last read try.

### PR DESCRIPTION
**Pull request by it-hms**
This update resolves a bug where the check for Circularized file looks back 1 minute in the Event File. This was changed to use absolute time and only look back to the most recent request. This error cased multiple error flags and requests. 

Closes #48 

image
image

### CODE REVIEW

**Comment by alexjhawk** [comment]
@it-hms Unless two applicable reviewers are unavailable (out of the office, etc.), a second reviewer should always be requested.

File: CHANGELOG.md
Comment (alexjhawk) [inline CHANGELOG.md:1]: Release commit is missing required changes. See a previous release commit, such as https://github.com/hms-networks/sc-ewon-flexy-extensions-lib/commit/550a530a65d67cfa96c51355b29d612dcff9b4d8.

File: src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java
Comment (alexjhawk) [inline src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java:179]: Is it possible that an event itself can contain the character `;`? If so, we should use `QuoteSafeStringTokenizer` here instead. An example of this can be found in the historical data manager: https://github.com/hms-networks/sc-ewon-flexy-extensions-lib/blob/550a530a65d67cfa96c51355b29d612dcff9b4d8/src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java#LL408C5-L408C30

**Review by alexjhawk — CHANGES_REQUESTED** [review]

File: src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java
Comment (it-hms) [reply src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java:179]: I think the only circumstance it if a use does the following in basic
`logevent "blah;blah;"`
However, I will change to safe tokenizer to handle this case

Comment (it-hms) [reply src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java]: @alexjhawk  @oliver-walker-hms  testing complete.

Comment (it-hms) [reply src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java]: @alexjhawk  @oliver-walker-hms  testing complete.

File: src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java
Comment (alexjhawk) [inline src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java:184]:
```suggestion
    boolean eventFound = false;
    if (exporterFile.length() > 0) {
      final List eventFileLines = StringUtils.split(exporterFile, EVENT_FILE_LINE_DELIMITER);
      for (int x = 1; x < eventFileLines.size(); x++) {
        String line = (String) eventFileLines.get(x);
        if (checkEventFileLine(line, eventId)) {
          eventFound = true;
          break;
        }
      }
    }
    return eventFound;
```

Not sure how cleanly this suggestion will show, but we should avoid returning in the middle of a method where possible. It makes the method more difficult to follow because there is no longer only one entry and exit point.

I suggest we add a boolean such as `boolean eventFound = false` and then `return eventFound;` at the end of the method. If we find the event during the loop, then we can do `eventFound = false; break;`

File: src/main/java/com/hms_networks/americas/sc/extensions/package.html
Comment (alexjhawk) [inline src/main/java/com/hms_networks/americas/sc/extensions/package.html:8]:
```suggestion
@version 1.14.0
```
Version number incorrect

File: pom.xml
Comment (alexjhawk) [inline pom.xml:11]: This is more of a discussion, but what warrants this version being 1.14.0 versus 1.13.11? 

A bump in the major version number seems inappropriate here, as there is only a single bug fix and no new features. (Per the changelog)

**Review by alexjhawk — CHANGES_REQUESTED** [review]

File: src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java
Comment (it-hms) [reply src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java:153]: will change

File: src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java
Comment (it-hms) [reply src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java:169]: will change

File: src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java
Comment (it-hms) [reply src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java:184]: will change

File: CHANGELOG.md
Comment (alexjhawk) [inline CHANGELOG.md:1]: Typically the title for release commits would be in the following format:

`RELEASE: v1.13.11`

I just checked quickly and the title *shouldn't* affect any of our scripts/automation though.

**Review by alexjhawk — APPROVED** [review]
Looks good + ready after pending second approval from @oliver-walker-hms.

I only added one (optional) comment/suggestion.

File: src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java
Comment (oliver-walker-hms) [inline src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java:22]: File does not need to be capitalized. This is repeated throughout the code on this and the next commit

**Review by oliver-walker-hms — APPROVED** [review]

File: src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java
Comment (alexjhawk) [reply src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java:22]: Agreed, although relatively minor

**Pull request merged**
